tiledb/sm/fragment/fragment_identifier.cc

```diff
@@ -109,6 +107,19 @@ FragmentID::FragmentID(const URI& uri)
   }
 }
 
+bool FragmentID::has_fragment_name(const URI& uri) {
```

Comment:
1. Can't we call this new function in the `FragmentID` constructor to reduce code duplication?
2. Ideally we should unit test this static function.

Reply:
Added unit test. We can call the function in the constructor, but we would make extra string allocations and the amount of duplication is not that big.